tracer/src/Datadog.Trace.Tools.Runner/Checks/ProcessBasicCheck.cs

```diff
@@ -106,13 +110,44 @@ public static bool Run(ProcessInfo process)
 
             if (corEnable != "1")
             {
-                Utils.WriteWarning(WrongEnvironmentVariableFormat(corEnableKey, "1", corEnable));
+                Utils.WriteError(WrongEnvironmentVariableFormat(corEnableKey, "1", corEnable));
                 foundIssue = true;
             }
 
-            if (!CheckRegistry())
+            string corProfilerPathKey = runtime == ProcessInfo.Runtime.NetCore ? "CORECLR_PROFILER_PATH" : "COR_PROFILER_PATH";
+
+            if (process.EnvironmentVariables.TryGetValue(corProfilerPathKey, out var profilerPath))
             {
-                foundIssue = true;
+                if (!IsExpectedProfilerFile(profilerPath))
+                {
+                    Utils.WriteError(WrongProfilerEnvironment(corProfilerPathKey, profilerPath));
+                    foundIssue = true;
+                }
+
+                if (!File.Exists(profilerPath))
+                {
+                    Utils.WriteError(MissingProfilerEnvironment(corProfilerPathKey, profilerPath));
+                    foundIssue = true;
+                }
+            }
+            else
+            {
+                if (!RuntimeInformation.IsOSPlatform(System.Runtime.InteropServices.OSPlatform.Windows))
+                {
+                    Utils.WriteError(EnvironmentVariableNotSet(corProfilerPathKey));
+                    foundIssue = true;
+                }
+            }
+
+            foundIssue &= !CheckProfilerPath(process, runtime == ProcessInfo.Runtime.NetCore ? "CORECLR_PROFILER_PATH_32" : "COR_PROFILER_PATH_32");
+            foundIssue &= !CheckProfilerPath(process, runtime == ProcessInfo.Runtime.NetCore ? "CORECLR_PROFILER_PATH_64" : "COR_PROFILER_PATH_64");
```

Comment:
Looks like you can _almost_ use `CheckProfilerPath()` above for `COR[ECLR]_PROFILER_PATH` (except for a "not Windows" check). Should we try to consolidate them?

Reply:
Yeah I got lazy here 😛  Done in https://github.com/DataDog/dd-trace-dotnet/pull/2482/commits/79f4d5e0e821a0cf45803c14842803934b2f908a

Reply:
Thanks!